pkg/image/api/helper.go

```diff
@@ -81,10 +86,26 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 		ref.Name = repoParts[0]
 		ref.Tag = tag
 		ref.ID = id
-		return ref, nil
+		break
 	default:
 		return ref, fmt.Errorf("the docker pull spec %q must be two or three segments separated by slashes", spec)
 	}
+
+	ns := ref.Namespace
+	if len(ns) == 0 {
+		ns = DockerDefaultNamespace
+	}
+	repo := ns + "/" + ref.Name
+
+	// Length checking
+	if len(ns) < DockerMinNamespaceLength {
```

Comment:
You can call this if you want https://github.com/docker/distribution/blob/master/registry/api/v2/names.go#L78